lib/jnpr/junos/utils/fs.py

```diff
@@ -268,30 +270,34 @@ def _decode(fs):
     # directory_usage - filesystem directory usage
     # -------------------------------------------------------------------------
 
-    def directory_usage(self, path="."):
+    def directory_usage(self, path=".", depth=0):
         """
         Returns the directory usage, similar to the unix "du" command.
 
-        :returns: approximate directory usage, including subdirectories, in bytes
+        :returns: dict of directory usage, including subdirectories if depth > 0
         """
-        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth="0")
+        BLOCK_SIZE = 512
 
-        used_space = rsp.findtext("directory/used-space")
+        rsp = self._dev.rpc.get_directory_usage_information(path=path, depth=str(depth))
 
-        if used_space is None:
-            return None
+        if rsp.findtext("directory/used-space") is None:
+            # Likely, no such directory
+            raise RpcError(rsp=rsp)
 
-        used_space = used_space.strip()
+        dirs = rsp.xpath("//directory")
+        result = {}
 
-        multiplier = {
-            'B': 1,
-            'K': 1024,
-            'M': 1024**2,
-            'G': 1024**3,
-            'T': 1024**4,
-        }
+        for directory in dirs:
```

Comment:
@stacywsmith findall('//directory') fails with error,

```
  File "/usr/lib/python3/dist-packages/lxml/_elementpath.py", line 246, in _build_path_iterator
    raise SyntaxError("cannot use absolute path on element")
SyntaxError: cannot use absolute path on element
```

that's why I was using xpath() initially. However findall('.//directory') works, so I will use it.